graylog2-server/src/main/java/org/graylog/scheduler/DBJobTriggerService.java

```diff
@@ -464,4 +479,33 @@ public void updateLockedJobTriggers() {
         DBUpdate.Builder update = DBUpdate.set(FIELD_LAST_LOCK_TIME, now);
         db.updateMulti(query, update);
     }
+
+    public void updateProgress(JobTriggerDto trigger, int progress) {
+        final Query query = DBQuery.is(FIELD_ID, trigger.id());
+        final DBUpdate.Builder update = DBUpdate.set(FIELD_PROGRESS, progress);
+        final int n = db.update(query, update).getN();
+        if (n != 1) {
+            LOG.warn("Progress update for trigger <{}> returned unexpected write result count <{}>", trigger, n);
+        }
+    }
+
+    public Optional<JobTriggerDto> cancelTriggerWithQuery(Query query) {
+        final DBUpdate.Builder update = DBUpdate.set(JobTriggerDto.FIELD_IS_CANCELLED, true);
+
+        return Optional.ofNullable(db.findAndModify(query, update));
+    }
+
+    public boolean triggerWasCancelled(JobTriggerDto trigger) {
+        final JobTriggerDto reloadedTrigger = db.findOneById(getId(trigger));
+        if (reloadedTrigger == null) {
+            return false;
+        }
```

Comment:
It could also be that the user doesn't have the right permissions (expressed in the query) to cancel the trigger.
We usually don't expose any information if an object is not found.

Reply:
ah, sorry. i confused this with the cancelTriggerByQuery() method. 
I could throw an exception, but I'm gonna ignore it in the consumer either way, so I don't think it makes much of a difference.
Your call :-)